### PR TITLE
tgui wires; refactor wire datums; more to come

### LINKED ISSUE #12301: syndicate pizza bomb cannot be stacked ontop of other pizzaboxes
i tried to put a syndicate pizza bomb ontop of a stack of other pizzas but it would just not work

### LINKED ISSUE #14749: Pizza boxes can't hold custom made pizzas

### PR DESCRIPTION
**Pull request by neersighted**
:cl:
experiment: Refactor wires; port wire interface to tgui.
add: Wire colors are now fully randomized.
del: Remove pizza bombs, as the code and sprites were terrible.
add: Add pizza bomb cores, which can be combined with a pizza box to make a pizza bomb.
/:cl:

Fixes #14749
Fixes #12301

image

### CODE REVIEW

**Comment by GunHog** [comment]
He just keeps going!!

**Comment by Iamgoofball** [comment]
@GunHog wires have been on the todo list for awhile because we have plans for the wire UI

**Comment by optimumtact** [comment]
Plus those things mentioned are quite significant overhauls

also plus having lots of simple ui's makes it easier for newbs to look at them to figure out how it works

**Comment by neersighted** [comment]
Yeah @GunHog a lot of things require significant refactors or total rewrites; and a lot of UIs are interdependent (see records consoles and PDAs). I plan to remove and rewrite PDAs from scratch; thus, big UIs will likely be post-freeze.

**Comment by neersighted** [comment]
@WJohn thanks again for the prompt sprites

**Comment by JJRcop** [comment]
:snowboarder: 
You're surfing on the freeze using tgui, I think the pizza box thing violates

**Comment by neersighted** [comment]
@JJRcop it maintains existing functionality; doesn't add anything new (except allowing custom pizzas which is a bug); just factors out the explosive into a bomb core... The bomb core is just there for code purposes primarily. Before, if you tried to remove the core from a pizza bomb you got a regular bomb core, is this is an improvement.

**Comment by JJRcop** [comment]
Well of course I'm no where near the arbiter of this stuff, I just wanted you to make sure you watch the lines, it seems like you're getting a little bit closer with this one.

File: code/modules/assembly/doorcontrol.dm [hunk 1]
```diff
@@ -47,15 +49,14 @@
 	for(var/obj/machinery/door/airlock/D in airlocks)
 		if(D.id_tag == src.id)
 			if(specialfunctions & OPEN)
-				spawn(0)
-					if(D)
-						if(D.density)	D.open()
-						else			D.close()
-					return
```
Comment (MrStonedOne) [inline code/modules/assembly/doorcontrol.dm]: This caused a bug fixed by https://github.com/tgstation/tgstation/pull/21125

@neersighted please don't remove spawns for no reason.

**Pull request merged**